tests/test_tag_release.py

```diff
@@ -0,0 +1,259 @@
+    }
+    # third request failed, so rollback the other two
+    assert responses.calls[3].request.url == "https://api.github.com/repos/edx/XBlock/git/refs/tags/tag-exists-some-repos"
+    assert responses.calls[3].request.method == 'DELETE'
+    assert responses.calls[4].request.url == "https://api.github.com/repos/edx/configuration/git/refs/tags/tag-exists-some-repos"
+    assert responses.calls[4].request.method == 'DELETE'
+
+
+def test_tag_repos_existing_tag_at_end_no_rollback(session, responses):
+    # creating a tag that already exists in edx-platform: we'll make sure
+    # that edx-platform is attempted *last*
+    ordered_commits = OrderedDict(
+        sorted(expected_commits.items(), key=lambda x: x[0], reverse=False)
+    )
+    with pytest.raises(RuntimeError) as excinfo:
+        tag_repos(
+            ordered_commits,
+            "tag-exists-some-repos",
+            session,
+            rollback_on_fail=False
+        )
+    assert len(responses.calls) == 3
+    assert responses.calls[0].request.url == "https://api.github.com/repos/edx/XBlock/git/refs"
+    xb_payload = json.loads(responses.calls[0].request.body)
+    assert xb_payload == {
+        "sha": "1a2b3c4d5e6f",
+        "ref": "refs/tags/tag-exists-some-repos",
+    }
+    assert responses.calls[1].request.url == "https://api.github.com/repos/edx/configuration/git/refs"
+    conf_payload = json.loads(responses.calls[1].request.body)
+    assert conf_payload == {
+        "sha": "12345deadbeef",
+        "ref": "refs/tags/tag-exists-some-repos",
+    }
+    assert responses.calls[2].request.url == "https://api.github.com/repos/edx/edx-platform/git/refs"
+    plat_payload = json.loads(responses.calls[2].request.body)
+    assert plat_payload == {
+        "sha": "deadbeef12345",
+        "ref": "refs/tags/tag-exists-some-repos",
+    }
+    assert "No rollback attempted" in str(excinfo.value)
+    assert "Reference already exists" in str(excinfo.value)
+    assert "Tags exist on the following repos: edx/XBlock, edx/configuration" in str(excinfo.value)
+
+
+def test_tag_repos_existing_tag_at_end_rollback_failure(session, responses):
+    # creating a tag that already exists in edx-platform: we'll make sure
+    # that edx-platform is attempted *last*
+    ordered_commits = OrderedDict(
+        sorted(expected_commits.items(), key=lambda x: x[0], reverse=False)
+    )
+
+    # when we try to delete the configuration tag, it will fail with a 500 error
+    responses.add(
+        responses.DELETE,
+        "https://api.github.com/repos/edx/configuration/git/refs/tags/tag-exists-some-repos",
+        status=500,
+    )
+
+    ### DANGER: ACCESSING PRIVATE APIS FOR RESPONSES LIBRARY ###
+    # grab the object we just created
+    last_responses_url_obj = responses._default_mock._urls[-1]
+    # move it to the front, so it matches *first*
+    responses._default_mock._urls.insert(0, last_responses_url_obj)
+    ### END PRIVATE API ACCESS FOR RESPONSES LIBRARY ###
```

Comment:
I don't know of a better way to do this, unfortunately. I'm open to suggestions.